var/spack/repos/builtin/packages/arrow/package.py

```diff
@@ -146,6 +146,11 @@ def cmake_args(self):
         args.append(self.define_from_variant("ARROW_WITH_ZLIB", "zlib"))
         args.append(self.define_from_variant("ARROW_WITH_ZSTD", "zstd"))
 
+        if not self.spec.satisfies("^re2"):
```

Comment:
This is for direct deps
```suggestion
        if not self.spec.dependencies("re2"):
```